frontend/src/components/viewers/MarkdownViewer.tsx

```diff
@@ -73,6 +74,16 @@ class MarkdownViewer extends Viewer<MarkdownViewerProps, any> {
     if (!this._config) {
       return null;
     }
+    if (this._config.markdownContent.length > getMaxMarkdownStrLength()) {
+      return (
+        <div className='markdown-viewer'>
+          <div data-testid='markdown-large-size-warning'>
+            This markdown is too large to render completely.
+          </div>
+          <Markdown>{this._config.markdownContent.substr(0, getMaxMarkdownStrLength())}</Markdown>
```

Comment:
I have searched for memorization, but didn't quite get a solution for it. I agree if we can implement some caching or in-place string edit, it will help the processing performance. Would you like to share some hint on how to do it?